src/Maestro/Maestro.Web/Api/v2018_07_16/Controllers/DefaultChannelsController.cs

```diff
@@ -52,7 +53,8 @@ public IActionResult List(string repository = null, string branch = null, int? c
 
             if (!string.IsNullOrEmpty(branch))
             {
-                query = query.Where(dc => dc.Branch == branch);
+                // Normalize the branch name to not include refs/heads
+                query = query.Where(dc => dc.Branch == IGitRepoExtension.NormalizeBranchName(branch));
```

Comment:
I thought this was the fix too. Perhaps I'm wrong but if this was the cause shouldn't it work when I used the API to filter mappings using branch = 'release/3.1.1xx'?

Reply:
I.e., when I use swagger to filter defaultchannels by channel == 'release/3.1.1xx' nothing is returned.

Reply:
But azdo doesn't pass branch names like that.  It always prefixes "refs/heads"

Reply:
Also, I think you need to put the `NormalizeBranchName` call outside the query. EntityFramework can't translate this call to SQL so its probably evaluating the query client side.